docs/src/modules/components/AppSearch.js

```diff
@@ -244,11 +239,11 @@ export default function AppSearch() {
   const macOS = window.navigator.platform.toUpperCase().indexOf('MAC') >= 0;
 
   return (
-    <div className={classes.root} style={{ display: desktop ? 'flex' : 'none' }}>
-      <div className={classes.search}>
+    <RootDiv>
+      <SearchDiv>
         <SearchIcon />
-      </div>
-      <Input
+      </SearchDiv>
+      <StyledInput
         disableUnderline
```

Comment:
This seems not work, not sure why.

<img width="1163" alt="Screen Shot 2564-07-13 at 16 03 45" src="https://user-images.githubusercontent.com/18292247/125424089-ac4cb97c-e81f-4140-a581-46fbbb1cfbbb.png">

Reply:
Maybe this is the reason - https://github.com/mui-org/material-ui/pull/27150/files#r668927047

Reply:
fixed in [`914931a` (#27150)](https://github.com/mui-org/material-ui/pull/27150/commits/914931a917e4f133dfa776587cc888050f1c2aed)

Reply:
Seeeing the code for the input component, the reason that this is broken is because you are setting the `componentsProps` property here https://github.com/mui-org/material-ui/pull/27150/files#diff-5209c11ef3f383b7aa76d05670d753415ff4f429b4b5d698e765cc0f5d10c724R272

This is the code for the Input component:

![image](https://user-images.githubusercontent.com/4512430/125600028-e3e86146-8978-4d11-91f8-b1fee1dc65ec.png)

So, setting the `componentsProps` is overriding the existing ones. Seems like currently we do not support the `componentsProps` on the input component, so I would say we should move the `sx` prop defined currently in the `componentsProps` to the `StyledInput` using the `'& input'` selector.

Reply:
I'm fairly certain I got `componentsProps` from autocomplete. Might also be a typing issue.

> using the '& input' selector.

That's not an Ok customization story. If we're already struggling to migrate our little usage then the story is even worse in larger apps. This should be fixed before stable release.

Reply:
Agree. The problem is with the `Input` component and how the `componentsProps` are passed. We should fix it for sure (merge them with the `props.componentsProps`)